src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -524,6 +516,18 @@ private void AddFunctionDescriptors()
             {
                 _logger.AddingDescriptorProviderForLanguage(_workerRuntime);
                 _descriptorProviders.Add(new RpcFunctionDescriptorProvider(this, _workerRuntime, ScriptOptions, _bindingProviders, _functionDispatcher, _loggerFactory, _applicationLifetime));
+                AddCodelessDescriptor();
+            }
+        }
+
+        /// <summary>
+        /// Adds a DotNetFunctionDescriptorProvider to the list of descriptors if any function metadata has language set to "codeless" in it.
+        /// </summary>
+        private void AddCodelessDescriptor()
+        {
+            if (_functionMetadataManager.GetFunctionMetadata().Any(m => string.Equals(m?.Language, DotNetScriptTypes.Codeless, StringComparison.OrdinalIgnoreCase)))
```

Comment:
remove the null checks on metadata here and elsewhere. Make sure no nulls get into the collection when its built